docs/docs/anatomy.md

```diff
@@ -0,0 +1,254 @@
+---
+sidebar_position: 3
+---
+
+# Anatomy of an Equinox Service
+
+An Equinox service is composed of a few modules (namespaces).
+
+- **Stream**: tells us which category the service writes to and how the identity
+  of the stream is composed (and the reverse operations for when running Reaction logic)
+- **Events**: tells us which events the service writes
+- **Fold**: tells us how the events are folded into state
+- **Decide**: tells us which actions can be taken upon the state (resulting in
+  new events being written)
+- **Service**: the class that wraps the above into a cohesive domain service.
+- **Config**: tells us how to wire the service against concrete stores,
+  including important information like access strategies.
+
+Let's see what each of these modules looks like for a simple service for
+checking in and out of an appointment.
+
+## The Stream
+
+```ts
+export namespace Stream {
+  export const category = "AppointmentActuals"
+  export const streamId = StreamId.gen(AppointmentId.toString, UserId.toString)
+  export const decodeId = StreamId.dec(AppointmentId.parse, UserId.parse)
+  export const tryMatch = StreamName.tryMatch(category, decodeId)
+}
+```
+
+## The Events
+
+```ts
+export namespace Events {
+  // prettier-ignore
+  const date = z.string().datetime().transform((x) => new Date(x))
+  export const Timestamp = z.object({ timestamp: date })
+  export const ActualsOverridden = z.object({ checkedIn: date, checkedOut: date })
+  export type ActualsOverridden = z.infer<typeof ActualsOverridden>
+
+  export type Event =
+    | { type: "CheckedIn"; data: Timestamp }
+    | { type: "CheckedOut"; data: Timestamp }
+    | { type: "ActualsOverridden"; data: ActualsOverridden }
+
+  export const codec = Codec.upcast<Event>(
+    Codec.json(),
+    Codec.Upcast.from({
+      CheckedIn: Timstamp.parse,
+      CheckedOut: Timstamp.parse,
+      ActualsOverridden: ActualsOverridden.parse,
+    }),
+  )
+}
+```
+
+## The Fold
+
+```ts
+export namespace Fold {
+  import Event = Events.Event
+  export type State = { checkedIn?: Date; checkedOut?: Date }
+  export const initial: State = {}
+  export const evolve = (state: State, event: Event) => {
+    switch (event.type) {
+      case "CheckedIn":
+        return { ...state, checkedIn: event.data.timestamp }
+      case "CheckedOut":
+        return { ...state, checkedOut: event.data.timestamp }
+      case "ActualsOverridden":
+        return event.data
+    }
+  }
+}
+```
+
+## The Decide
```

Comment:
this reads wacko
And while its perfect for a small example, it odoes not show how you might subdivide it for multiple decisions

Reply:
This'll be the second item in the nav. Not sure if it's the right place for a full-blown complex thing. Don't want to scare people just yet 😅